src/modules/atommodel/primitives/RadixAddress.ts

```diff
@@ -124,7 +124,7 @@ export class RadixAddress implements RadixPrimitive {
     }
 
     public verify(data: Buffer, signature: RadixECSignature) {
-        return this.keyPair.verify(data, EC.Signature(signature))
+        return this.keyPair.verify(data, {r: signature.r.toString(), s: signature.s.toString()})
```

Comment:
The commit message says it all, I guess. I'd like some feedback on this alternative implementation.
Note! that even if it fixes the immediate error, the verification still fails:

```javascript
it('Signature of signed atom can be verified', () => {
    const atom = new RadixAtom()
    const identity = new RadixIdentityManager().generateSimpleIdentity()

    await identity.signAtom(atom)

    const signatureId = identity.address.getUID().toString()

    // Fails! The verification evaluates to boolean false.
    expect(identity.address.verify(
            atom.getHash(),
            atom.signatures[signatureId])
    ).to.equal(true)
}
```

I'm not sure why that is...

Reply:
Haha, well then it's not really a fix, is it? 

I kid, it's off to a good start, the problem it looks to me is that `RadixBytes:toString()` doesn't encode the value as the right kind of string, I think it wants hex.

But going through a string is an unnecessary detour, since it supposedly accepts
`// 4) object with two buffer properties (r and s)`
And luckily for you RadixBytes is just a wrapper for a Buffer.

So you can try `{r: signature.r.bytes, s: signature.s.bytes}`


If that doesn't work, we might need to reverse the `RadixECSignature.fromEllasticSignature` method, which sometimes pads the signature components to make it work with the node code. (It also occurs to me that it should be called `fromEllipticSignature`)

Reply:
Good feedback! I'll try it out immediately!